File: Release/WindowsPhone7/PhoneToolkitSample/Data/MoviesByCategory.vb
' (c) Copyright Microsoft Corporation.
' This source is subject to the Microsoft Public License (Ms-PL).
' Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
' All other rights reserved.

Imports System.Collections.ObjectModel

Namespace Data

    Public Class MoviesByCategory
        Inherits ObservableCollection(Of MoviesInCategory)
        Public Sub New()
            Dim sortedCategories As New List(Of String)(Movie.Categories)
            sortedCategories.Sort()

            For Each category In sortedCategories
                Dim group As New MoviesInCategory(category)
                Me.Add(group)

                For i = 0 To 4
                    group.Add(Movie.CreateRandom(category))
                Next i
            Next category
        End Sub
    End Class


    Public Class MoreCommand
        Implements ICommand

#Region "ICommand Members"

        Public Function CanExecute(ByVal parameter As Object) As Boolean Implements System.Windows.Input.ICommand.CanExecute
            Return True
        End Function

        Public Event CanExecuteChanged As EventHandler Implements System.Windows.Input.ICommand.CanExecuteChanged

        Public Sub Execute(ByVal parameter As Object) Implements System.Windows.Input.ICommand.Execute
            Dim group = TryCast(parameter, MoviesInCategory)
            If group IsNot Nothing Then
                group.Add(Movie.CreateRandom(group.Key))
            End If
        End Sub

#End Region

    End Class

End Namespace

File: Release/WindowsPhone71/PhoneToolkitSample/Data/MoviesByCategory.cs
// (c) Copyright Microsoft Corporation.
// This source is subject to the Microsoft Public License (Ms-PL).
// Please see http://go.microsoft.com/fwlink/?LinkID=131993 for details.
// All other rights reserved.

using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Windows.Input;

namespace PhoneToolkitSample.Data
{
    public class MoviesByCategory : ObservableCollection<MoviesInCategory>
    {
        public MoviesByCategory()
        {
            List<string> sortedCategories = new List<string>(Movie.Categories);
            sortedCategories.Sort();

            foreach (string category in sortedCategories)
            {
                MoviesInCategory group = new MoviesInCategory(category);
                this.Add(group);

                for (int i = 0; i < 5; ++i)
                {
                    group.Add(Movie.CreateRandom(category));
                }                
            }
        }
    }

    public class MoreCommand : ICommand
    {
        #region ICommand Members

        public bool CanExecute(object parameter)
        {
            return true;
        }

        public event EventHandler CanExecuteChanged;

        public void Execute(object parameter)
        {
            MoviesInCategory group = parameter as MoviesInCategory;
            if (group != null)
            {
                group.Add(Movie.CreateRandom(group.Key));
            }
        }

        #endregion
    }
}
